src/job.rs

```diff
@@ -131,88 +128,79 @@ impl Job {
             ok_to_continue = true;
         }
         if ok_to_continue {
-            let filename_s = self.param.get(Param::File).unwrap_or_default();
-            filename = unsafe { filename_s.strdup() };
-            if unsafe { strlen(filename) } == 0 {
-                warn!(context, 0, "Missing file name for job {}", self.job_id,);
-            } else if 0 != unsafe { dc_read_file(context, filename, &mut buf, &mut buf_bytes) } {
-                if let Some(recipients) = self.param.get(Param::Recipients) {
-                    let recipients_list = recipients
-                        .split("\x1e")
-                        .filter_map(|addr| match lettre::EmailAddress::new(addr.to_string()) {
-                            Ok(addr) => Some(addr),
-                            Err(err) => {
-                                eprintln!("WARNING: invalid recipient: {} {:?}", addr, err);
-                                None
+            if let Some(filename) = self.param.get(Param::File) {
+                if let Some(body) = dc_read_file_safe(context, filename) {
+                    if let Some(recipients) = self.param.get(Param::Recipients) {
+                        let recipients_list = recipients
+                            .split("\x1e")
+                            .filter_map(|addr| match lettre::EmailAddress::new(addr.to_string()) {
+                                Ok(addr) => Some(addr),
+                                Err(err) => {
+                                    eprintln!("WARNING: invalid recipient: {} {:?}", addr, err);
+                                    None
+                                }
+                            })
+                            .collect::<Vec<_>>();
+                        /* if there is a msg-id and it does not exist in the db, cancel sending.
+                        this happends if dc_delete_msgs() was called
+                        before the generated mime was sent out */
+                        let ok_to_continue1;
+                        if 0 != self.foreign_id {
+                            if 0 == unsafe { dc_msg_exists(context, self.foreign_id) } {
+                                warn!(
+                                    context,
+                                    0,
+                                    "Message {} for job {} does not exist",
+                                    self.foreign_id,
+                                    self.job_id,
+                                );
+                                ok_to_continue1 = false;
```

Comment:
can we not now better return instead of ok_to_continue1=false and remove all "if ok_to_continue" logic? 
the PR is anyway changing indentation sufficiently widely to make reviewing hard.  When removing all ok_to_continue here, it becomes easier to compare it to the C-version.

Reply:
Fine.